corehq/apps/userreports/reports/data_source.py

```diff
@@ -102,13 +106,22 @@ def get_data(self, slugs=None):
             ProgrammingError,
         ) as e:
             raise UserReportsError(e.message)
-        # arbitrarily sort by the first column in memory
-        # todo: should get pushed to the database but not currently supported in sqlagg
+        # TODO: Should sort in the database instead of memory, but not currently supported by sqlagg.
         try:
-            return sorted(ret, key=lambda x: x.get(
-                self.column_configs[0].column_id,
-                next(x.itervalues())
-            ))
+            # If a sort order is specified, sort by it.
+            if self._order_by:
+                for col in reversed(self._order_by):
+                    ret.sort(
+                        key=lambda x: x.get(col[0], None),
+                        reverse=col[1] == "DESC"
```

Comment:
(trivial) I prefer a constant variable instead of referring to string literals.  btw you can just go ahead and ignore that advice when reviewing my most recent PR